### PR TITLE
Added Delete Cluster Feature For Targets

### PR DESCRIPTION
**Pull request by oumkale**
Signed-off-by: Oum Kale <oum.kale@mayadata.io>

## Proposed changes

This PR is adding a Delete cluster feature for targets. We can delete cluster from Cluster Info screens as well as Table now.

Added Delete Cluster
gql integration
Updated mutation, queries, and modal  
Back Button 

## Types of changes

What types of changes does your code introduce to Litmus? Put an `x` in the boxes that apply
- [x] New feature (non-breaking change which adds functionality)
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices applies)

## Checklist

Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.
- [ ] I have read the [CONTRIBUTING](https://github.com/litmuschaos/litmus/blob/master/CONTRIBUTING.md) doc
- [ ] I have signed the commit for DCO to be passed.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have added necessary documentation (if appropriate)

## Dependency
- Please add the links to the dependent PR need to be merged before this (if any).

### CODE REVIEW

File: litmus-portal/frontend/src/views/ChaosWorkflows/BrowseCluster/index.tsx [hunk 1]
```diff
@@ -73,12 +74,16 @@ const BrowseCluster = () => {
...
         project_id: selectedProjectID,
+        cluster_type: 'external',
       },
       fetchPolicy: 'cache-and-network',
       pollInterval: 3000,
     }
   );
 
+  // Apollo mutation to delete the selected Target Cluster
+  const [deleteCluster] = useMutation<DeleteCluster>(DELETE_CLUSTER, {});
```
Comment (Saranya-jena) [inline litmus-portal/frontend/src/views/ChaosWorkflows/BrowseCluster/index.tsx:85]:
```suggestion
  const [deleteCluster] = useMutation<DeleteCluster>(DELETE_CLUSTER);
```

File: litmus-portal/frontend/src/views/ChaosWorkflows/BrowseCluster/TableData.tsx [hunk 2]
```diff
@@ -62,6 +78,59 @@ const TableData: React.FC<TableDataProps> = ({ data }) => {
...
+                        setOpen(false);
+                      }}
+                    >
+                      <> {t('targets.modalDelete.no')}</>
+                    </ButtonOutline>
+
+                    <ButtonFilled
+                      isDisabled={false}
+                      isPrimary
+                      // styles={classes.error}
```
Comment (Saranya-jena) [inline litmus-portal/frontend/src/views/ChaosWorkflows/BrowseCluster/TableData.tsx:122]: Can you please remove unused comments?

File: litmus-portal/frontend/src/components/Targets/ClusterInfo/index.tsx [hunk 3]
```diff
@@ -97,18 +121,53 @@ const ClusterInfo: React.FC<ClusterVarsProps> = ({ location }) => {
...
+                          setOpen(false);
+                        }}
+                      >
+                        <> {t('targets.modalDelete.no')}</>
+                      </ButtonOutline>
+
+                      <ButtonFilled
+                        isDisabled={false}
+                        isPrimary
+                        // styles={classes.error}
```
Comment (Saranya-jena) [inline litmus-portal/frontend/src/components/Targets/ClusterInfo/index.tsx:160]: remove unused comments

File: litmus-portal/frontend/src/components/Targets/ClusterInfo/index.tsx [hunk 4]
```diff
@@ -1,14 +1,20 @@
...
 import ButtonOutline from '../../Button/ButtonOutline';
 // import BrowseWorkflow from '../TargetHome/BrowseWorkflow';
 import useStyles from './styles';
 import Scaffold from '../../../containers/layouts/Scaffold';
 import TargetCopy from '../TargetCopy';
-import { Cluster } from '../../../models/graphql/clusterData';
+import { Cluster, DeleteCluster } from '../../../models/graphql/clusterData';
 import { LocationState } from '../../../models/routerModel';
+import { DELETE_CLUSTER } from '../../../graphql';
+
```
Comment (Saranya-jena) [inline litmus-portal/frontend/src/components/Targets/ClusterInfo/index.tsx:14]: Can you remove this line change?

File: litmus-portal/frontend/src/components/Targets/ClusterInfo/index.tsx [hunk 5]
```diff
@@ -21,7 +27,12 @@ const ClusterInfo: React.FC<ClusterVarsProps> = ({ location }) => {
   const { data } = location.state;
   const classes = useStyles();
   const link: string = data.token;
-  const handleClick = () => {
+
+  const [deleteCluster] = useMutation<DeleteCluster>(DELETE_CLUSTER, {});
```
Comment (Saranya-jena) [inline litmus-portal/frontend/src/components/Targets/ClusterInfo/index.tsx:31]:
```suggestion
  const [deleteCluster] = useMutation<DeleteCluster>(DELETE_CLUSTER);
```

**Review by Saranya-jena — CHANGES_REQUESTED** [review]

File: litmus-portal/frontend/src/components/Targets/ConnectTarget/index.tsx [hunk 6]
```diff
@@ -71,6 +68,10 @@ const ConnectTarget = () => {
       project_id: selectedProjectID,
       cluster_type: 'external',
       agent_scope: 'cluster',
+      agent_namespace: 'litmus',
+      serviceaccount: '',
+      agent_sa_exists: false,
+      agent_ns_exists: false,
```
Comment (ishangupta-ds) [inline litmus-portal/frontend/src/components/Targets/ConnectTarget/index.tsx:74]: why are these hardcoded? cc: @rajdas98 @gdsoumya @oumkale 

agent_scope: 'cluster',
agent_namespace: 'litmus',

**Review by ishangupta-ds — CHANGES_REQUESTED** [review]

File: litmus-portal/frontend/src/components/Targets/ConnectTarget/index.tsx [hunk 7]
```diff
@@ -71,6 +68,10 @@ const ConnectTarget = () => {
       project_id: selectedProjectID,
       cluster_type: 'external',
       agent_scope: 'cluster',
+      agent_namespace: 'litmus',
```
Comment (ishangupta-ds) [inline litmus-portal/frontend/src/components/Targets/ConnectTarget/index.tsx:71]: why are these hardcoded? cc: @rajdas98 @gdsoumya @oumkale 

agent_scope: 'cluster',
agent_namespace: 'litmus',

**Review by ishangupta-ds — CHANGES_REQUESTED** [review]

**Review by ishangupta-ds — APPROVED** [review]
LGTM.

**Review by amityt — APPROVED** [review]

**Review by Jonsy13 — APPROVED** [review]

**Pull request merged**